seisgo/noise.py

```diff
@@ -384,55 +379,218 @@ def do_correlation(sfile,win_len,step,maxlag,cc_method='xcorr',acorr_only=False,
+            for iiS in range(ndata):
+                # get index right for auto/cross correlation
+                istart=iiS
+                src=fftdata[iiS].net+"."+fftdata[iiS].sta
+                # if acorr_only:iend=np.minimum(iiS+ncomp,ndata)
+                # if xcorr_only:istart=np.minimum(iiS+ncomp,ndata)
+                #-----------now loop II for each receiver B----------
+                for iiR in range(istart,iend):
+                    # if v:print('receiver: %s %s' % (fftdata[iiR].net,fftdata[iiR].sta))
+                    rcv=fftdata[iiR].net+"."+fftdata[iiR].sta
+                    if (acorr_only and src==rcv) or (xcorr_only and src != rcv) or (not acorr_only and not xcorr_only):
+                        if fftdata[iiS].data is not None and fftdata[iiR].data is not None:
+                            if v:print('receiver: %s %s' % (fftdata[iiR].net,fftdata[iiR].sta))
+                            corrdata=correlate(fftdata[iiS],fftdata[iiR],maxlag,method=cc_method,substack=substack,
+                                                smoothspect_N=smoothspect_N,substack_len=substack_len,
+                                                maxstd=maxstd)
+
+                            if corrdata.data is not None:
+                                if output_structure.lower() == "raw" or output_structure.lower() == "r":
+                                    corrdata.to_asdf(file=outfile,v=v)
+                                elif output_structure.lower() == "source" or output_structure.lower() == "s":
+                                    corrdata.to_asdf(file=os.path.join(fhead,corrdata.net[0]+'.'+corrdata.sta[0],ftail),v=v)
+                                elif output_structure.lower() == "station-pair" or output_structure.lower() == "sp":
+                                    netsta_pair = corrdata.net[0]+'.'+corrdata.sta[0]+'_'+\
+                                                    corrdata.net[1]+'.'+corrdata.sta[1]
+                                    corrdata.to_asdf(file=os.path.join(fhead,netsta_pair,ftail),v=v)
+                                elif output_structure.lower() == "station-component-pair" or output_structure.lower() == "scp":
+                                    netsta_pair = corrdata.net[0]+'.'+corrdata.sta[0]+'_'+\
+                                                    corrdata.net[1]+'.'+corrdata.sta[1]
+                                    chan_pair = corrdata.chan[0]+'_'+corrdata.chan[1]
+                                    corrdata.to_asdf(file=os.path.join(fhead,netsta_pair,chan_pair,ftail),v=v)
+                                else:
+                                    raise ValueError(output_structure + " is not recoganized. must be one of "+\
+                                            str(output_o)+" or "+str(output_o_short))
 
     # create a stamp to show time chunk being done
     ftmp.write('done')
     ftmp.close()
 
-    return ndata
+    return ndata,ttt1,ttt2,ttt3,ttt4,ttt5,ttt6
```

Comment:
can we wrap the returned ttt* in a list, to simplify the output when people call the function?